CMakeLists.txt

```diff
@@ -1,10 +1,64 @@
-cmake_minimum_required(VERSION 2.8)
+cmake_minimum_required(VERSION 3.0)
 project(termcolor)
 
-if(CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang")
-  set(CMAKE_CXX_FLAGS "-Werror -Wall -pedantic")
+# define target to install and link tests against
+add_library(termcolor INTERFACE)
+target_include_directories(termcolor
+  INTERFACE $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>)
+
+# ALIAS same as in configure file
+# the ALIAS can be used to create examples which use the same syntax as a client
+# application, which uses `find_package(libdivide CONFIG)`
+# create the alias libdivide::libdivide
+add_library(termcolor::termcolor ALIAS termcolor)
+
+# add compiler flags for test targets only for GCC and Clang
+target_compile_options(${PROJECT_NAME} INTERFACE
+  $<BUILD_INTERFACE:
+    $<$<OR:$<CXX_COMPILER_ID:GNU>,$<CXX_COMPILER_ID:Clang>>:
+      -Werror -Wall -pedantic>>
+  )
+
+option(ENABLE_TESTING "enable testing with CTEST" ON)
+if(ENABLE_TESTING)
+  enable_testing()
+  add_executable(test_${CMAKE_PROJECT_NAME} test/test.cpp)
+  add_test(test_termcolor test_${CMAKE_PROJECT_NAME})
+  target_link_libraries(test_${CMAKE_PROJECT_NAME} PRIVATE termcolor::termcolor)
+  add_custom_target(run COMMAND test_${CMAKE_PROJECT_NAME})
 endif()
 
-include_directories(${termcolor_SOURCE_DIR}/include)
-add_executable(test_${CMAKE_PROJECT_NAME} test/test.cpp)
-add_custom_target(run COMMAND test_${CMAKE_PROJECT_NAME})
+# create install target
+install(
+  TARGETS termcolor
+  EXPORT termcolor-targets
+  INCLUDES DESTINATION include # set include path for installed library target
+  )
+install(
+  FILES include/termcolor/termcolor.hpp
+  DESTINATION include/termcolor
+  )
+
+# Include module for fuctions
+# - 'write_basic_package_version_file'
+# - 'configure_package_config_file'
+include(CMakePackageConfigHelpers)
+
+# generate and install termcolor-config.cmake file
+# Configure '<PROJECT-NAME>-config.cmake'
+configure_package_config_file(
+  "cmake/config.cmake.in"
+  "${CMAKE_CURRENT_BINARY_DIR}/generated/termcolor-config.cmake"
+  INSTALL_DESTINATION "lib/cmake/termcolor"
+)
+# install config file
+install(
+  FILES "${CMAKE_CURRENT_BINARY_DIR}/generated/termcolor-config.cmake"
+  DESTINATION "lib/cmake/termcolor"
+)
+# install targets file
+install(
+  EXPORT "termcolor-targets"
+  NAMESPACE "termcolor::"
```

Comment:
I'm not an expert in CMake and packaging in C++, so pardon my silly question. Do we really need a namespace here? `termcolor` unlikely to ever have more than one target which means namespace seems redundant here. What do you think?

Reply:
please ask questions! The only silly thing to do is to NOT ask :)

We don't strickly need it, but the namespace is modern cmake convention and holds information for the programmer using your project. The difference between `termcolor` and `termcolor::termcolor` is the simple fact that you (and cmake) instantly get the information, that `termcolor::termcolor` is an imported target and not built by your project

While explaining I got the feeling I maybe should remove the usage of the alias when creating the test

If something is still unclear please ask :)